num_enum_derive/src/lib.rs

```diff
@@ -595,7 +664,11 @@ pub fn derive_from_primitive(input: TokenStream) -> TokenStream {
 #[proc_macro_derive(TryFromPrimitive, attributes(num_enum))]
 pub fn derive_try_from_primitive(input: TokenStream) -> TokenStream {
     let enum_info: EnumInfo = parse_macro_input!(input);
-    let krate = Ident::new(&get_crate_name(), Span::call_site());
+    let krate = enum_info.crate_path.clone().unwrap_or_else(|| {
+        syn::LitStr::new(&format!("::{}", get_crate_name()), Span::call_site())
+            .parse_with(syn::Path::parse_mod_style)
+            .unwrap()
+    });
```

Comment:
Hmm, given the code duplicate, I wonder if `get_crate_name()` shouldn't be taking a `Option<Path>` parameter and being the one emitting the actual path?